new-dti-website/components/courses/Timeline.tsx

```diff
@@ -0,0 +1,222 @@
+ * Each event includes a title, date, and time. The date string should be formatted with the month (abbreviated), day, and
+ * optionally, the time (formatted as `hh:mm AM/PM`). If no time is provided, the default will be 12:00 AM.
+ * If no Year is provided, the default will be the Current Year
+ *
+ * @param props - Contains:
+ *   - `events`: An array of event objects. Each object should contain:
+ *     - `title`: The name of the event.
+ *     - `date`: The date of the event in the format `MMM DD, YYYY` (e.g., "Feb 19, 2024"). MONTH NOT REQUIRED TO BE ABBREVIATED both Feb and February are valid
+ *     - `time`: (Optional) The time of the event in the format `hh:mm AM/PM` (e.g., "12:00 PM EST").
+ *   - `currentDate`: A `Date` object representing the current date and time, used to calculate the progress through the timeline.
+ */
+export default function Timeline({ events, currentDate }: TimelineProps) {
+  const firstEventRef = useRef<HTMLDivElement | null>(null);
+  const lastEventRef = useRef<HTMLDivElement | null>(null);
+  const [lineLength, setLineLength] = useState(0);
+  const [isMobile, setIsMobile] = useState(false);
+
+  /**
+   * Parses an event's date and time to a `Date` object.
+   *
+   * @remarks
+   * This function converts a date string (formatted as `MMM DD`) and an optional time string (formatted as `hh:mm AM/PM`)
+   * into a JavaScript `Date` object. If no time is provided, the default time is 12:00 AM. The year is automatically
+   * set to the current year if no year is provided, so ensure that the date values are formatted properly.
+   *
+   * @param dateStr - The date string of the event in the format `MMM DD, YYYY` (e.g., "Feb 19, 2024"). MONTH NOT REQUIRED TO BE ABBREVIATED
+   * @param timeStr - (Optional) The time of the event in the format `hh:mm AM/PM` (e.g., "12:00 PM EST").
+   *
+   * @returns A `Date` object corresponding to the event's date and time.
+   *
+   * @example
+   * ```ts
+   * const eventDate = parseEventDate("Feb 19", "7:30 PM");
+   * ```
+   */
+  const parseEventDate = (dateStr: string, timeStr: string) => {
+    const currentYear = new Date().getFullYear();
+    const date = new Date(`${dateStr} ${currentYear} ${timeStr || '12:00 AM'}`);
+    return date;
+  };
+
+  /**
+   * Calculates the percentage of progress through the events based on the current date.
+   *
+   * @remarks
+   * This function sorts the events by date, calculates the total time span between the first
+   * and last events, and determines how far along the current date is within that time span.
+   * The result is a percentage that is used to fill the progress line on the timeline.
+   *
+   * @returns A number representing the percentage of progress (from 0 to 100).
+   *
+   * @example
+   * ```ts
+   * const progressPercentage = getProgressPercentage();
+   * ```
+   */ const getProgressPercentage = () => {
```

Comment:
super duper minor nit: could you put this on a separate line so it's not on the same line as the docstring?